lighthouse-core/audits/audit.js

```diff
@@ -175,6 +174,28 @@ class Audit {
 
 module.exports = Audit;
 
+/**
+ * @typedef {Object} Audit.ScoringModes
+ * @property {'numeric'} NUMERIC
+ * @property {'binary'} BINARY
+ */
+
+/**
+ * @typedef {Audit.ScoringModes[keyof Audit.ScoringModes]} Audit.ScoringModeValues
+ */
+
+/**
+ * @typedef {Object} Audit.Meta
+ * @property {string} name
+ * @property {string} description
+ * @property {string} helpText
+ * @property {Array<string>} requiredArtifacts
+ * @property {string=} failureDescription
```

Comment:
how do you feel about doing `foo|undefined` instead of `foo=`? i can almost promise i'll forget what `foo=` means in another 3 weeks.

Reply:
I don't mind the explicitness, especially because it's only allowed in some circumstances, but on the other hand it can get pretty unwieldy and there's no jsdoc equivalent to the nice `?:` on the ts side

Reply:
k can we do `foo|undefined`?

Reply:
oh but there is! http://usejsdoc.org/tags-property.html :)

```js
/**
 * @property {Array<string>} requiredArtifacts
 * @property {string} [failureDescription]
 */
```

Reply:
that...does not seem better than `=` :P

Reply:
it seems slightly better to me, but not as good as ? or |undefined

Reply:
you guys are crazy, optional is part of the type not the name!

luckily we can all agree on good old boring dependable `|undefined`

Reply:
> you guys are crazy, optional is part of the type not the name!

but the semantics are different!

`|undefined` implies the key is always there but just potentially with undefined, the `[prop]` syntax more correctly suggests that the property name is optional rather than the type of the value is sometimes undefined :)

Reply:
typescript treats `{type=}` and `[propName]` the same, but the distinction from `|undefined` *is* important...and I was wrong before about `|undefined`. tsc isn't going to let us use it anyways with `--strict` on.

The difference is erased when you get something that is this type, but not when declaring that something is one of these types...e.g. with `@property {boolean|undefined} informative` you better have a property `informative` that's one of the two. A missing `informative` won't suffice, so the property is no longer optional.

I like this reasoning of `[prop]`, it offers a nice way of expressing the defined/not-defined vs value/undefined mixup of JS, so let's go with that.